src/test-util.ts

```diff
@@ -32,3 +34,195 @@ export function b(
   }
   return utf8.encode(s);
 }
+
+export function defineTestFunctions(self: Window & typeof globalThis): void {
+  // Mocha functions aren't available when importing from a new page.
+  if (typeof self.suiteSetup !== 'function') {
+    // eslint-disable-next-line @typescript-eslint/no-empty-function
+    const noop = function () {};
+
+    self.setup = noop;
+    self.teardown = noop;
+    self.suiteSetup = noop;
+    self.suiteTeardown = noop;
+
+    // eslint-disable-next-line @typescript-eslint/no-empty-function
+    const suite = function () {};
+    suite.only = noop;
+    suite.skip = noop;
+    self.suite = suite as unknown as Mocha.SuiteFunction;
+
+    // eslint-disable-next-line @typescript-eslint/no-empty-function
+    const test = function () {};
+    test.only = noop;
+    test.skip = noop;
+    test.retries = noop;
+    self.test = test as unknown as Mocha.TestFunction;
+  }
+}
+
+defineTestFunctions(self);
+
+type Response = {
+  tabId: string;
+  id: string;
+  result?: unknown;
```

Comment:
optional unknown seems a bit strange. Just do unknown?

There is a new feature in TS4.4 which allows it to  distinguish between missing properties vs `| undefined`. We are not using that yet.

Reply:
If I omit the `?`, I get the below error later in the file. Anything else to try here?

```
Property 'result' is missing in type '{ tabId: any; id: any; }' but required in type 'Response'.`
test-util.ts[66, 3]: 'result' is declared here.
```

Reply:
haha

OK TypeScript, you win this time.